src/toil/job.py

```diff
@@ -67,13 +69,13 @@ def __init__(self, requirements, unitName, displayName=None, jobName=None):
         memory = requirements.get('memory')
         disk = requirements.get('disk')
         preemptable = requirements.get('preemptable')
-        if unitName is not None:
+        if unitName:
             assert isinstance(unitName, (str, bytes))
-        if jobName is not None:
+        if jobName:
```

Comment:
Fair enough, in this situation it's checking that it's type `string`, which should always be true when it's not `''` or `None`, so it seemed safe.  But it's a fair point that I should be more conservative because you're right, it's not always equivalent.  Will change the majority of these back and talk to you about any others.

Reply:
I've changed them all back, except for the ones in `src/toil/job.py`.  If you still feel it would be better to change back, I will.  I do not think the ones in `src/toil/job.py` will affect anything though.

Reply:
Thanks for changing them. I think it's OK to leave these, because it's pretty obvious that both `''` and `None` can be treated the same way without breaking anything, even though it may technically be a slight change in behavior. No one will really rely on their job to be named `''`.